delivery_multi_destination/models/delivery_carrier.py

```diff
@@ -21,8 +21,30 @@ class DeliveryCarrier(models.Model):
             ('one', 'One destination'),
             ('multi', 'Multiple destinations'),
         ],
-        default="one", required=True,
+        compute="_compute_destination_type",
+        inverse="_inverse_destination_type",
+        store=True,
     )
+    delivery_type = fields.Selection(
+        selection_add=[("base_on_destination", "Based on Destination")]
+    )
+
+    @api.depends("delivery_type")
+    def _compute_destination_type(self):
+        for carrier in self:
+            if carrier.delivery_type == "base_on_destination":
+                carrier.destination_type = "multi"
+            else:
+                carrier.destination_type = "one"
+
+    def _inverse_destination_type(self):
+        for carrier in self:
+            # Switch to multi
+            if carrier.destination_type == "multi":
+                carrier.delivery_type = "base_on_destination"
+            # Switch away from multi
+            elif carrier.delivery_type == "base_on_destination":
+                carrier.delivery_type = "fixed"
```

Comment:
It may be a mistake here. Shouldn't it be something like this ?
```python
if carrier.destination_type == "multi":
    carrier.delivery_type = "base_on_destination"
elif carrier.destination_type == "one":
    carrier.delivery_type = "fixed"

Reply:
Are you sure that tests succeed ?

Reply:
Yes, maybe this merits a comment. I wrote it like this for … spaghetti reasons. Writing the spaghetti code did things to my brain.

Presently, `destination_type` kind of behaves like a boolean. It is, however, not a boolean. It's a selection field. This is probably a design error, but I didn't/don't want to address that in this PR.

With that in mind, your proposed code snippet would look a little like this:

```python
if carrier.destination_type == "multi":
    carrier.delivery_type = "base_on_destination"
# destination type is 'one' or something else
else:
    carrier.delivery_type = "fixed"
```

But that didn't make me happy. If we imagine a scenario where there are two non-"multi" destination types, then switching back and forth between those would re-set the delivery type to "fixed" (from a potentially valid value) every time, for no reason! Rather, we only want to re-set the delivery type when it would otherwise be invalid (i.e., when it's set to 'base_on_destination').

Of course, a module that introduces a new destination type would need to override the compute and inverse functions, but by doing things this way, we annoy downstream as little as possible.

Reply:
The tests run on my machine, and the tests succeed on Travis as well, just not on Python 3.5.

Reply:
Created a comment in a fixup commit.